addons/mail/static/src/core/common/message_model.js

```diff
@@ -89,6 +89,9 @@ export class Message extends Record {
         if ("author" in data) {
             this.author = data.author;
         }
+        if ("author_id" in data && data["author_id"].length) {
+            this.author = { id: data.author_id[0], name: data.author_id[1] };
+        }
```

Comment:
No, adapt formatter using `author_id` so that it is properly formatted for discuss models.
`{ id, name, type: "partner" }`

I'm simplifying discuss models insert to just become data from server => immediately inserted without any custom code. The formatter should work immediately with `Message.insert()` soon, so keep same shape as `author` relation needs, which is something like `{id, type}` at the very least.